### DISCUSSION TITLE
Loading Atomrefs in Pretrained DimeNet mode

### DISCUSSION THREAD

**Discussion by Frank-LIU-520** [pyg-team/pytorch_geometric#2656] [Q&A]
In the **qm9_pretrained_dimenet.py** of the example codes, there read  [# DimeNet uses the atomization energy for targets U0, U, H, and G.](url)  in line 12. But there is no way to find out how did DimeNet load and use these atomrefs in QM9 class. Can anybody tell me where I can trace how the atomrefs were loaded and used in the DimeNet training routine?

`import os.path as osp

import torch

from torch_geometric.datasets import QM9
from torch_geometric.data import DataLoader
from torch_geometric.nn import DimeNet

path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'QM9')
dataset = QM9(path)

**# DimeNet uses the atomization energy for targets U0, U, H, and G.**
idx = torch.tensor([0, 1, 2, 3, 4, 5, 6, 12, 13, 14, 15, 11])
dataset.data.y = dataset.data.y[:, idx]`

**Comment by rusty1s** [comment]
In the [QM9 dataset](https://pytorch-geometric.readthedocs.io/en/latest/modules/datasets.html#torch_geometric.datasets.QM9), there exists both properties `U_0` and `U_0^ATOM`. While some papers train against the `U_0` property (target 7), DimeNet trains against U_0^ATOM (target 12). This is also mentioned in the DimeNet paper: "We only use the atomization energy for U0, U, H, and G, i.e.  subtract the atomic reference energies, which are constant per atom type."

**Reply by Frank-LIU-520** [reply]
Thanks for the answer. Got it.
